### PR TITLE
More tweaks on fine tuning

### PR DESCRIPTION
**Pull request by Peilin-Yang**

### CODE REVIEW

File: src/main/resources/fine_tuning/models.yaml [hunk 1]
```diff
@@ -1,5 +1,34 @@
 ---
 models:
+  bm25:
+    name: bm25
+    fixed_params: "-bm25"
```
Comment (lukuang) [inline src/main/resources/fine_tuning/models.yaml]: redundant `fix_params` for `bm25` and `ql`?

File: src/main/python/fine_tuning/run_batch.py [hunk 2]
```diff
@@ -152,7 +152,7 @@ def del_method_related_files(method_name):
     parser.add_argument('--anserini_root', default='', help='Anserini path')
     parser.add_argument('--collection', required=True, help='the collection key in yaml')
     parser.add_argument('--basemodel', default='bm25', choices=['bm25', 'ql'], help='the ranking model')
-    parser.add_argument('--model', default='axiom', choices=['axiom', 'rm3', 'sdm'], help='the higher level model')
+    parser.add_argument('--model', default='axiom', choices=['bm25', 'ql', 'axiom', 'rm3', 'bm25+axiom', 'bm25+rm3'], help='the higher level model')
```
Comment (lukuang) [inline src/main/python/fine_tuning/run_batch.py]: Is it a good idea to mix model with basemodel? I think here is the reason why all your result files start with "bm25"..

Comment (lukuang) [reply src/main/python/fine_tuning/run_batch.py]: Or we abandon basemodel all together?

Comment (lukuang) [reply src/main/python/fine_tuning/run_batch.py]: Like what you did with `bm+rm3`?

File: src/main/python/fine_tuning/effectiveness.py [hunk 3]
```diff
@@ -128,25 +128,31 @@ def add_up_all_optimal(self, json_data):
...
         for fn in os.listdir(effectiveness_root):
             basemodel, model, metric = fn.split('_')
             if metric not in metrics:
                 continue
+            if metric not in per_topic_oracle:
+                per_topic_oracle[metric] = {}
             with open(os.path.join(effectiveness_root, fn)) as f:
                 for real_metric, all_performance in json.load(f).items():
-                    all_optimal = self.add_up_all_optimal(all_performance)
+                    all_optimal = self.add_up_all_optimal(all_performance, per_topic_oracle[metric])
```
Comment (lukuang) [inline src/main/python/fine_tuning/effectiveness.py]: Shouldn't here be `real_metric`?

File: src/main/python/fine_tuning/effectiveness.py [hunk 4]
```diff
@@ -128,25 +128,31 @@ def add_up_all_optimal(self, json_data):
                         if param_value not in params_dist[param_name]:
                             params_dist[param_name][param_value] = 0
                         params_dist[param_name][param_value] += 1
+                if qid not in per_topic_oracle_with_metric:
+                    per_topic_oracle_with_metric[qid] = json_data[qid]['max']['value']
+                else:
+                    per_topic_oracle_with_metric[qid] = max(json_data[qid]['max']['value'], per_topic_oracle_with_metric[qid])
         return round(sum_optimal/n, 5), params_dist
```
Comment (lukuang) [inline src/main/python/fine_tuning/effectiveness.py]: `params_dist` is returned but not used

File: src/main/java/io/anserini/analysis/EnglishAnalyzerStemming.java [hunk 5]
```diff
@@ -0,0 +1,63 @@
+package io.anserini.analysis;
```
Comment (lintool) [inline src/main/java/io/anserini/analysis/EnglishAnalyzerStemming.java]: add license header?

File: src/main/java/io/anserini/analysis/EnglishAnalyzerStemming.java [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+import org.apache.lucene.analysis.*;
+import org.apache.lucene.analysis.en.EnglishPossessiveFilter;
+import org.apache.lucene.analysis.en.KStemFilter;
+import org.apache.lucene.analysis.en.PorterStemFilter;
+import org.apache.lucene.analysis.miscellaneous.SetKeywordMarkerFilter;
+import org.apache.lucene.analysis.standard.StandardAnalyzer;
+import org.apache.lucene.analysis.standard.StandardFilter;
+import org.apache.lucene.analysis.standard.StandardTokenizer;
+
+public class EnglishAnalyzerStemming extends StopwordAnalyzerBase {
```
Comment (lintool) [inline src/main/java/io/anserini/analysis/EnglishAnalyzerStemming.java]: How about `EnglishStemmingAnalyzer`? Makes for sense to me?

File: src/main/python/fine_tuning/effectiveness.py [hunk 7]
```diff
@@ -132,27 +132,25 @@ def add_up_all_optimal(self, json_data, per_topic_oracle_with_metric):
...
-            if metric not in metrics:
-                continue
-            if metric not in per_topic_oracle:
-                per_topic_oracle[metric] = {}
             with open(os.path.join(effectiveness_root, fn)) as f:
                 for real_metric, all_performance in json.load(f).items():
-                    all_optimal = self.add_up_all_optimal(all_performance, per_topic_oracle[metric])
+                    if real_metric not in per_topic_oracle:
+                        per_topic_oracle[real_metric] = {}
+                    all_optimal = self.add_up_all_optimal(all_performance, per_topic_oracle[real_metric])
```
[the remark this answers is not in the document]
Comment (Peilin-Yang) [reply src/main/python/fine_tuning/effectiveness.py]: yea, they are currently not used in verification but can be potentially used in the further analysis

**Pull request merged**